data/transaction/construction_test.go

```diff
@@ -0,0 +1,178 @@
+		Nonce:    92,
+		Value:    stringToBigInt("123456789000000000000000000000"),
+		RcvAddr:  getPubkeyOfAddress(t, "erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx"),
+		SndAddr:  getPubkeyOfAddress(t, "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"),
+		GasPrice: 1000000000,
+		GasLimit: 100000,
+		Data:     []byte("for the spaceship"),
+		ChainID:  []byte("local-testnet"),
+		Version:  1,
+	}
+
+	tx.Signature = signTransaction(t, aliceSeedHex, tx)
+	require.Equal(t, "39938d15812708475dfc8125b5d41dbcea0b2e3e7aabbbfceb6ce4f070de3033676a218b73facd88b1432d7d4accab89c6130b3abe5cc7bbbb5146e61d355b03", hex.EncodeToString((tx.Signature)))
+
+	data := marshalTransactionToProto(tx)
+	require.Equal(t, "085c120e00018ee90ff6181f3761632000001a208049d639e5a6980d1cd2392abcce41029cda74a1563523a202f09641cc2618f82a200139472eff6886771a982f3083da5d421f24c29181e63888228dc81ca60d69e1388094ebdc0340a08d064a11666f722074686520737061636573686970520d6c6f63616c2d746573746e65745801624039938d15812708475dfc8125b5d41dbcea0b2e3e7aabbbfceb6ce4f070de3033676a218b73facd88b1432d7d4accab89c6130b3abe5cc7bbbb5146e61d355b03", hex.EncodeToString((data)))
+
+	txHash := computeTransactionHash(tx)
+	require.Equal(t, "e4a6048d92409cfe50f12e81218cb92f39966c618979a693b8d16320a06061c1", hex.EncodeToString((txHash)))
+}
+
+func stringToBigInt(input string) *big.Int {
+	result := big.NewInt(0)
+	_, _ = result.SetString(input, 10)
+	return result
+}
+
+func getPubkeyOfAddress(t *testing.T, address string) []byte {
+	pubkeyConverter, _ := pubkeyConverter.NewBech32PubkeyConverter(32)
+	pubkey, err := pubkeyConverter.Decode(address)
+	require.Nil(t, err)
+	return pubkey
+}
+
+func signTransaction(t *testing.T, senderSeedHex string, tx *Transaction) []byte {
+	keyGenerator := signing.NewKeyGenerator(ed25519.NewEd25519())
+	signer := &singlesig.Ed25519Signer{}
+
+	senderSeed, err := hex.DecodeString(senderSeedHex)
+	require.Nil(t, err)
+
+	privateKey, err := keyGenerator.PrivateKeyFromByteArray(senderSeed)
+	require.Nil(t, err)
+
+	jsonToSign := marshalTransactionToJSON(tx)
+	signature, err := signer.Sign(privateKey, jsonToSign)
+	require.Nil(t, err)
+
+	return signature
+}
+
+func marshalTransactionToJSON(tx *Transaction) []byte {
+	frontendTransaction := &FrontendTransaction{
+		Nonce:            tx.Nonce,
+		Value:            tx.Value.String(),
+		Receiver:         getAddressOfPubkey(tx.RcvAddr),
+		Sender:           getAddressOfPubkey(tx.SndAddr),
+		GasPrice:         tx.GasPrice,
+		GasLimit:         tx.GasLimit,
+		SenderUsername:   tx.SndUserName,
+		ReceiverUsername: tx.RcvUserName,
+		Data:             tx.Data,
+		ChainID:          string(tx.ChainID),
+		Version:          tx.Version,
+		Options:          tx.Options,
+	}
+
+	marshalizer := &marshal.JsonMarshalizer{}
+	jsonToSign, _ := marshalizer.Marshal(frontendTransaction)
+	return jsonToSign
+}
+
+func marshalTransactionToProto(tx *Transaction) []byte {
+	marshalizer := &marshal.GogoProtoMarshalizer{}
```

Comment:
instead of using hardcoded marshalizers, hashers and converters, why not have a var section for them?

Reply:
Fixed.